app/src/main/java/com/github/studydistractor/sdp/procrastinationActivity/FirebaseDistractionStat.kt

```diff
@@ -0,0 +1,126 @@
+package com.github.studydistractor.sdp.procrastinationActivity
+
+import android.util.Log
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+
+class FirebaseDistractionStat : DistractionStatInterface {
+
+    private val pathFeedback = "Feedback"
+    private val pathLikes = "Likes"
+    private val pathDislikes = "Dislikes"
+    private val pathTags = "Tags"
+    private val db = FirebaseDatabase.getInstance()
+
+    override fun fetchDistractionFeedback(did: String): SnapshotStateList<String> {
```

Comment:
@lipefree made an interesting comment about how to do the "fetch function" (i.e `fetchUser` in FirebaseUserService). Here is the link of the comment: https://github.com/StudyDistractor/sdp/pull/99#discussion_r1172460954

It is not complicated and you can check the code example [here](https://github.com/StudyDistractor/sdp/blob/3be678df2e5b8bc4fc8d285711f5f3811b4d9304/app/src/main/java/com/github/studydistractor/sdp/user/FirebaseUserService.kt)